dnf/subject.py

```diff
@@ -122,10 +122,16 @@ def get_best_selector(self, sack, forms=None):
             kwargs['form'] = forms
         nevra = first(self.subj.nevra_possibilities_real(sack, **kwargs))
         if nevra:
-            sltr = dnf.selector.Selector(sack)
-            s = self._nevra_to_selector(sltr, nevra)
-            if len(s.matches()) > 0:
-                return s
+            if nevra._has_just_name():
+                sltr = dnf.selector.Selector(sack)
+                s = sltr.set(provides=nevra.name)
```

Comment:
the `if` could apply only to `s` assignment, the other statements are the same in both code paths.